app/controllers/course_sessions_controller.rb

```diff
@@ -34,11 +34,18 @@ def create
     end
   end
 
-  # PATCH/PUT /lectures/1
-  # PATCH/PUT /lectures/1.json
+  # PATCH/PUT /course_sessions/1
+  # PATCH/PUT /course_sessions/1.json
   def update
     respond_to do |format|
       if @course_session.update(course_session_params)
+        # send an email here if the session is open for applications
+        if @course_session.admissions?
+          WaitingList.current.each do |waiting_list|
```

Comment:
nit pick, may want to move this to e.g. `@course_session.notify_wait_list!` to keep controller slim

Reply:
Good point, will do!

Reply:
After a bit more thought, I don't think this belongs on CourseSession, as it's not related, but I can put it on WaitingList, where it makes a lot of sense.